examples_tests/52.SystemTest/main.cpp

```diff
@@ -126,45 +126,60 @@ class WindowEventCallback : public IWindow::IEventCallback
 	WindowEventCallback(core::smart_refctd_ptr<InputSystem>&& inputSystem, system::logger_opt_smart_ptr&& logger) : m_inputSystem(std::move(inputSystem)), m_logger(std::move(logger)) {}
 
 private:
-	void onWindowShown_impl() override 
+	bool onWindowShown_impl() override 
 	{
 		m_logger.log("Window Shown");
+		return true;
 	}
-	void onWindowHidden_impl() override 
+	bool onWindowHidden_impl() override
 	{
 		m_logger.log("Window hidden");
+		return true;
 	}
-	void onWindowMoved_impl(int32_t x, int32_t y) override
+	bool onWindowMoved_impl(int32_t x, int32_t y) override
 	{
 		m_logger.log("Window window moved to { %d, %d }", system::ILogger::ELL_WARNING, x, y);
+		return true;
 	}
-	void onWindowResized_impl(uint32_t w, uint32_t h) override
+	bool onWindowResized_impl(uint32_t w, uint32_t h) override
 	{
 		m_logger.log("Window resized to { %u, %u }", system::ILogger::ELL_DEBUG, w, h);
+		return true;
 	}
-	void onWindowMinimized_impl() override
+	bool onWindowMinimized_impl() override
 	{
 		m_logger.log("Window minimized", system::ILogger::ELL_ERROR);
+		return true;
 	}
-	void onWindowMaximized_impl() override
+	bool onWindowMaximized_impl() override
 	{
 		m_logger.log("Window maximized", system::ILogger::ELL_PERFORMANCE);
+		return true;
 	}
-	void onGainedMouseFocus_impl() override
+	bool onGainedMouseFocus_impl() override
 	{
 		m_logger.log("Window gained mouse focus", system::ILogger::ELL_INFO);
+		return true;
 	}
-	void onLostMouseFocus_impl() override
+	bool onLostMouseFocus_impl() override
 	{
 		m_logger.log("Window lost mouse focus", system::ILogger::ELL_INFO);
+		return true;
 	}
-	void onGainedKeyboardFocus_impl() override
+	bool onGainedKeyboardFocus_impl() override
 	{
 		m_logger.log("Window gained keyboard focus", system::ILogger::ELL_INFO);
+		return true;
 	}
-	void onLostKeyboardFocus_impl() override
+	bool onLostKeyboardFocus_impl() override
 	{
 		m_logger.log("Window lost keyboard focus", system::ILogger::ELL_INFO);
+		return true;
+	}
+	bool onWindowClosed_impl() override
```

Comment:
can you prevent/disallow the closing of a window?

If no, a boolean doesn't make sense

Reply:
yep, you definitely can

Reply:
ok nice.